utils/dbutils.py

```diff
@@ -0,0 +1,151 @@
+
+    if debug:
+        df_gemView.info()
+        print("Read {0} rows from view {1}".format(df_gemView.shape[0],view))
+        pass
+
+    if vfatList is not None:
+        # First warn the user which VFATs are *not* found
+        if len(vfatList) != df_gemView.shape[1]:
+            printYellow("Length of returned view does not match length of input vfat List")
+            vfatsNotFound = [ str(hex(chipId)).strip('L') for chipId in vfatList if str(hex(chipId)).strip('L') not in list(df_gemView['vfat3_ser_num'])]
+            printYellow("VFATs not found: {0}".format(vfatsNotFound))
+            pass
+        
+        # Then add a 'vfatN' column to the output df; this increases row # to len(vfatList)
+        df_gemView = joinOnVFATSerNum(vfatList,df_gemView)
+        
+        pass
+
+    return df_gemView
+
+def getVFAT3CalInfo(vfatList, fromProd=True, debug=False):
+    """
+    Gets from GEM_VFAT3_PROD_SUMMARY_V_RH view a subset of data that is necessary
+    for VFAT calibration.  Specifically a pandas dataframe will be returned with
+    only the following columns:
+
+        ['vfatN','vfat3_ser_num', 'vfat3_barcode', 'iref', 'adc0m', 'adc1m', 'adc0b', 'adc1b', 'cal_dacm', 'cal_dacb']
+
+    vfatList    - list of VFAT Chip ID's.
+    fromProd    - If True (False) the view is taken from the Production (Development) DB
+    debug       - Prints additional info if true
+    """
+
+    df_vfatCalInfo = getVFAT3ProdSumView(vfatList, fromProd, debug)
+
+    return df_vfatCalInfo[['vfatN','vfat3_ser_num', 'vfat3_barcode', 'iref', 'adc0m', 'adc1m', 'adc0b', 'adc1b', 'cal_dacm', 'cal_dacb']]
+
+def getVFAT3ConfView(vfatList, fromProd=True, debug=False):
+    """
+    Gets the GEM_VFAT3_CHIP_CONF_V_RH view in the GEM DB for a list of input VFATs.
+
+    Returns a pandas dataframe object storing the data read from the DB
+
+    vfatList    - list of VFAT Chip ID's.
+    fromProd    - If True (False) the view is taken from the Production (Development) DB
+    debug       - Prints additional info if true
+    """
+
+    return getGemDBView("GEM_VFAT3_CHIP_CONF_V_RH",vfatList,fromProd,debug)
+
+def getVFAT3ProdSumView(vfatList, fromProd=True, debug=False):
+    """
+    Gets the GEM_VFAT3_PROD_SUMMARY_V_RH view in the GEM DB for a list of input VFATs.
+
+    Returns a pandas dataframe object storing the data read from the DB
+
+    vfatList    - list of VFAT Chip ID's.
+    fromProd    - If True (False) the view is taken from the Production (Development) DB
+    debug       - Prints additional info if true
+    """
+
+    return getGemDBView("GEM_VFAT3_PROD_SUMMARY_V_RH",vfatList,fromProd,debug)
+
+def getVFATFilter(vfatList):
+    """
+    Returns a string that can be used as a filter in an SQL query.
+
+    vfatList - list of VFAT Chip ID's
+    """
+
+    strRetFilter = " WHERE ("
+    for idx, vfatID in enumerate(vfatList):
+        if idx == 0:
+            strRetFilter += " data.VFAT3_SER_NUM='{0}'".format(str(hex(vfatID)).strip('L'))
```

Comment:
Why not use native string formatting? (also elsewhere)
`{:x}` instead of `str(hex(num)).strip('L')`
(add appropriate padding and prefix if required)

Reply:
Was not really familiar with this; I've updated it.